Sources/StreamChatUI/ChatMessageList/ChatMessageListCollectionView.swift

```diff
@@ -5,13 +5,112 @@
 import StreamChat
 import UIKit
 
+/// Protocol that adds delegate methods specific for `ChatMessageListCollectionView`
+public protocol ChatMessageListCollectionViewDelegate: UICollectionViewDelegate {
+    /// Get date for item at given indexPath
+    /// - Parameters:
+    ///   - collectionView: CollectionView requesting date
+    ///   - indexPath: IndexPath that should be used to get date
+    func collectionView(_ collectionView: ChatMessageListCollectionView, dateForItemAt indexPath: IndexPath) -> Date?
+}
+
 /// The collection view that provides convenient API for dequeuing `_СhatMessageCollectionViewCell` instances
 /// with the provided content view type and layout options.
-open class ChatMessageListCollectionView: UICollectionView {
+open class ChatMessageListCollectionView: UICollectionView, Customizable, ComponentsProvider {
+    // Class does not use any ExtraData generic so we just need to make compiler happy
+    public typealias ExtraData = NoExtraData
+    
     private var identifiers: Set<String> = .init()
     
     open var needsToScrollToMostRecentMessage = false
     open var needsToScrollToMostRecentMessageAnimated = false
+    
+    /// View used to display date of currently displayed messages
+    open lazy var dateView: ChatMessageListDateView = {
+        let dateView = components.messageList.dateView.init()
+        dateView.isHidden = true
+        return dateView.withoutAutoresizingMaskConstraints
+    }()
```

Comment:
Maybe we can name this `scrollOverlayView`? This view should be visible only when the collection view is scrolling.